code/modules/mining/boulder_processing.dm

```diff
@@ -319,10 +319,41 @@
 	circuit = /obj/item/circuitboard/machine/refinery
 	usage_sound = 'sound/machines/mining/refinery.ogg'
 
+	/// Reagents that we can use to wash the boulders
+	var/list/allowed_reagents = list(/datum/reagent/water, /datum/reagent/lube)
```

Comment:
You mentioned a third reagent, could probably do it there. This can also be an index list(/datum/reagent/water = 6.8, /datum/reagent/lube) or whatever if you want extra data for efficiency